### PR TITLE
Feature invitation link

### PR DESCRIPTION
**Pull request by fedescarpa**

### CODE REVIEW

File: app/routes.rb [hunk 1]
```diff
@@ -144,3 +145,8 @@ def notify_upsert_exam(exam_id)
 require_relative './routes/notifications'
 require_relative './routes/suggestions'
 
+options '*' do
+  response.headers['Allow'] = settings.allow_methods.map { |it| it.to_s.upcase }.join(',')
```
Comment (flbulgarelli) [inline app/routes.rb]: Why do we need cross origin support? Isn't this app supposed to already implement it?

File: lib/classroom/models/invitation.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  field :code, type: String, default: -> { Invitation.generate_code }
+  field :course_slug, type: String
+  field :expiration_date, type: Time
+  belongs_to :course
+
+  before_validation :check_if_expired!
+
+  def check_if_expired!
+    if expired?
+      raise Exception.new "Must be in future"
```
Comment (flbulgarelli) [inline lib/classroom/models/invitation.rb]: `raise 'Must be in future'`

File: app/routes.rb (hunk 1 above)
Comment (fedescarpa) [reply app/routes.rb]: Yes I need it. I don't know why it fails, but adding the option endpoint doen't fail any more.

File: lib/classroom/models/invitation.rb (hunk 2 above)
Comment (fedescarpa) [reply lib/classroom/models/invitation.rb]: Roger!

**Pull request merged**